src/plone/restapi/services/users/add.py

```diff
@@ -299,6 +299,13 @@ def update_password(self, data):
         # Reset the password with a reset token
         if reset_token:
             try:
+                err = registration_tool.testPasswordValidity(new_password)
+                if err is not None:
+                    return self._error(
+                        400,
+                        "Invalid password",
+                        _(err),
```

Comment:
It seems that sometimes `testPasswordValidity` returns an [already translated string](https://github.com/plone/Products.CMFPlone/blob/8d33738b058b2e9d8d73566d4a161a3379b79c81/Products/CMFPlone/RegistrationTool.py#L225), sometimes it returns a [msgid](https://github.com/plone/Products.CMFPlone/blob/8d33738b058b2e9d8d73566d4a161a3379b79c81/Products/CMFPlone/RegistrationTool.py#L198). Then you need to do a test to see what is returned. This should be fixed in Plone.

Reply:
uhnn but you are not passing the `confirm` parameter. Shouldn't you pass it here?

Reply:
> It seems that sometimes `testPasswordValidity` returns an [already translated string](https://github.com/plone/Products.CMFPlone/blob/8d33738b058b2e9d8d73566d4a161a3379b79c81/Products/CMFPlone/RegistrationTool.py#L225), sometimes it returns a [msgid](https://github.com/plone/Products.CMFPlone/blob/8d33738b058b2e9d8d73566d4a161a3379b79c81/Products/CMFPlone/RegistrationTool.py#L198). Then you need to do a test to see what is returned. This should be fixed in Plone.

I would expect `_(...)` to do the right thing in any case? +1 for fixing this in the `RegistrationTool`.

> uhnn but you are not passing the `confirm` parameter. Shouldn't you pass it here?

There's a sensible default and we do not have a password confirmation in this case.

> Another weird thing is we have an update_password in add.py. Shouldn't this be on update.py?

Maybe, but imho that's for another ticket.

Reply:
@wesleybl I suggest you open the tickets yourself.